packages/smart-wallet/src/walletFactory.js

```diff
@@ -143,8 +141,9 @@ export const makeAssetRegistry = assetPublisher => {
  * }} privateArgs
  * @param {import('@agoric/vat-data').Baggage} baggage
  */
-export const start = async (zcf, privateArgs, baggage) => {
+export const prepare = async (zcf, privateArgs, baggage) => {
   const { agoricNames, board, assetPublisher } = zcf.getTerms();
+  const upgrading = baggage.has('walletsByAddress');
```

Comment:
consider: inlining the `has` to the conditional below. This is a valid signal of "upgrading" but we don't need that general signal.

Reply:
I moved the declaration next to the use.

Fully inlined, it seemed to need a comment. Might as well let the code speak for itself.